otter/convergence/service.py

```diff
@@ -172,7 +172,8 @@ def update_cache(group, servers, lb_nodes, now):
         sd = thaw(server.json)
         if is_autoscale_active(server, lb_nodes):
             sd["_is_as_active"] = True
-        server_dicts.append(sd)
+        if server.state != ServerState.DELETED or include_deleted:
```

Comment:
@manishtomar: The logic would have to be `if not (include_deleted and server.state == ServerState.DELETED)`, which seems more complicated. If you're just saying it'd be more readable if the conditions should be swapped, I can do that.